config/lib/mongoose.js

```diff
@@ -24,25 +25,28 @@ module.exports.connect = function (cb) {
 
   mongoose.Promise = config.db.promise;
 
-  var db = mongoose.connect(config.db.uri, config.db.options, function (err) {
-    // Log Error
-    if (err) {
-      console.error(chalk.red('Could not connect to MongoDB!'));
-      console.log(err);
-    } else {
+  var options = _.merge(config.db.options || {}, { useMongoClient: true });
 
+  mongoose
+    .connect(config.db.uri, options)
+    .then(function (connection) {
       // Enabling mongoose debug mode if required
       mongoose.set('debug', config.db.debug);
 
       // Call callback FN
-      if (cb) cb(db);
-    }
-  });
+      if (cb) cb(connection.db);
+    })
+    .catch((err) => {
```

Comment:
I'd love to see bigger discussion about this, actually. This topic just came up at Trustroots so I might open an issue here just to get some food for thought.

I don't think it's a good idea to introduce mixed coding style in PR like this.

It's more annoying to me to see mixed style rather than old school JS.

I'd love to see meanjs move on to ES6 but it might be better done with some strategy and thought and build tools (eslint) could probably be upgraded first.

WDYT?